### PR TITLE
buffer store with oob check

### PR DESCRIPTION
**Pull request by asleepzzz**
buffer store with oob check

### CODE REVIEW

**Comment by whchung** [comment]
@asleepzzz please focus on `f32` for now, and keep on improving the codes so it runs end-to-end, plus consider the cases enlarging along `gemmM` and `gemmN` dimensions.

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h [hunk 1]
```diff
@@ -6255,10 +6256,82 @@ struct ThreadwiseCopyRewritePattern
         else
           destLowerIndices = destUpperIndices;
 
+        bool toEmitOOBStoreCheckLogic = false;
+        SmallVector<unsigned, 2> oobStoreCheckDims;
+        if (destLowerIndices.size() == 5) {
+          for (unsigned iter = 0; iter < destTransform.getNumResults();
```
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: please give an example to explain logic here.

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h [hunk 2]
```diff
@@ -6255,10 +6256,82 @@ struct ThreadwiseCopyRewritePattern
...
+                                           b.getIntegerType(32));
+          auto iv4 = b.create<IndexCastOp>(loc, destLowerIndices[4],
+                                           b.getIntegerType(32));
+          // 2G ,INT MAX Value = 2147483647, use 2147483648 as offset and buffer
+          // store do nothing
+          Value oobAddrOp =
+              b.create<ConstantIntOp>(loc, 2147483648, b.getIntegerType(32));
+
+          auto coordLast = destLowerIndices[4];
+          auto sevenConstantOp = b.create<ConstantIndexOp>(loc, 7);
```
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: what is this for?

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h [hunk 3]
```diff
@@ -6255,10 +6295,70 @@ struct ThreadwiseCopyRewritePattern
...
+
         // Store to dest.
         // Issue scalar store.
-        b.create<StoreOp>(loc, convertedScalarValue, op.dest(),
-                          destLowerIndices);
+        if (toEmitOOBStoreCheckLogic) {
+          Value withinStoreBoundsOp =
+              b.create<ConstantIntOp>(loc, 1, b.getIntegerType(1));
+
+          auto iv0 = b.create<IndexCastOp>(loc, destLowerIndices[0],
```
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: IV has its own meaning, induction variable, in a compiler. please rename these variables.

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h [hunk 4]
```diff
@@ -6255,10 +6295,70 @@ struct ThreadwiseCopyRewritePattern
...
+
+          auto thenBuilder = ifWithinBoundsOp.getThenBodyBuilder();
+          thenBuilder.create<scf::YieldOp>(loc,
+                                           ValueRange{iv0, iv1, iv2, iv3, iv4});
+          auto elseBuilder = ifWithinBoundsOp.getElseBodyBuilder();
+          elseBuilder.create<scf::YieldOp>(
+              loc, ValueRange{iv0, iv1, oobAddrOp, iv3, iv4});
+          // loc, ValueRange{iv0,iv1,oobAddrOp,iv3,iv4});
+
+          b.create<gpu::MubufStoreOp>(loc, convertedScalarValue, op.dest(),
```
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: @asleepzzz  per discussion, we are suspecting the symptoms you saw is potentially due to wrapping around when we linearize the offset.

to continue look into the issue, please do:

1) change host code so it populates predefined non-zero constants onto the output tensor.
2) make sure tuning parameters are set consistently.
3) change GPU dialect syntax of `MubufStoreOp` so we supply another operand to decide whether to add a 2G offset.
4) change GPU->ROCDL lowering pipeline to use the operand and add a 2G offset in case needed.

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h [hunk 5]
```diff
@@ -6255,10 +6295,70 @@ struct ThreadwiseCopyRewritePattern
...
+
+          auto iv0 = b.create<IndexCastOp>(loc, destLowerIndices[0],
+                                           b.getIntegerType(32));
+          auto iv1 = b.create<IndexCastOp>(loc, destLowerIndices[1],
+                                           b.getIntegerType(32));
+          auto iv2 = b.create<IndexCastOp>(loc, destLowerIndices[2],
+                                           b.getIntegerType(32));
+
+          Value oobAddrOp =
+              b.create<ConstantIntOp>(loc, 2147483648, b.getIntegerType(32));
```
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: per discussion, we don't need this Value here anymore. it would sit inside GPU->ROCDL conversion process instead.

**Comment by whchung** [comment]
@asleepzzz Per discussion, let's change the scope of this PR a little bit:

- revise `gpu.buffer_store` syntax to adopt a new offset operand.
- change GPU->ROCDL logic for `gpu.buffer_store` to add the 2G offset in case needed.
- implement `toCheckOOBStoreLogic` in `threadwise_copy` and `threadwise_copy_v2`, BUT leave the flag as `false`.
- implement OOB store logic per current PR.

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h (hunk 5 above)
Comment (asleepzzz) [reply mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: we still need it , I pass it as offset

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h (hunk 1 above)
Comment (asleepzzz) [reply mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: the logic of toEmitOOBStoreCheck remove now ,
toEmitOOBStoreCheckLogic = false;
will discuss with letao how to enable it

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h [hunk 6]
```diff
@@ -519,7 +519,8 @@ struct Conv2DRewritePattern : public OpRewritePattern<T> {
     // filter : K & CRS , if CRS is under 64 or 32
     // we pad CRS to 32 or 64, then mlir can do gemm
     // we add more one transform to do pad
-    if (convOpType == miopen::ConvOpType::Conv2DOpType && gemmKExtra > 0) {
+    if (convOpType == miopen::ConvOpType::Conv2DOpType &&
```
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: Per discussion, logic concerning padding along K/M/N dimensions shall Not be included in this PR

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h [hunk 7]
```diff
@@ -614,13 +612,39 @@ struct Conv2DRewritePattern : public OpRewritePattern<T> {
 
       if (gemmMExtra > 0) {
         if (arg0TargetLayoutName1 == "gemmM") {
-          isFilterPad = false;
-          isGemmDim1Pad = false;
+          isFilterPad = true;
```
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: Changes here shall be excluded from this PR

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h [hunk 8]
```diff
@@ -6507,10 +6547,117 @@ struct ThreadwiseCopyRewritePattern
         SmallVector<Value, 8> destLowerIndices =
             layeredDestIndices[layeredDestIndices.size() - 1];
 
+        // we do not  implement the logic of toEmitOOBStoreCheck yet
+        bool toEmitOOBStoreCheckLogic = false;
+        SmallVector<unsigned, 2> oobStoreCheckDims;
+        if (destLowerIndices.size() == 5) {
```
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: This if is also better be commented out

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h [hunk 9]
```diff
@@ -6507,10 +6547,117 @@ struct ThreadwiseCopyRewritePattern
...
+
         // Store to dest.
         // Issue scalar store.
-        b.create<StoreOp>(loc, convertedScalarValue, op.dest(),
-                          destLowerIndices);
+        if (toEmitOOBStoreCheckLogic) {
+          SmallVector<Value, 8> destLowerStoreOP;
+          SmallVector<Value, 8> destLowerStoreOOBOp;
+
+          auto dstIndex0 = b.create<IndexCastOp>(loc, destLowerIndices[0],
```
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: Use a smallVector instead of supplying 5 variables

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h [hunk 10]
```diff
@@ -6507,10 +6547,117 @@ struct ThreadwiseCopyRewritePattern
...
+          destLowerStoreOP.push_back(dstIndex0);
+          destLowerStoreOP.push_back(dstIndex1);
+          destLowerStoreOP.push_back(dstIndex2);
+          destLowerStoreOP.push_back(dstIndex3);
+          destLowerStoreOP.push_back(dstIndex4);
+          destLowerStoreOOBOp = destLowerStoreOP;
+          // 2G ,INT MAX Value = 2147483647, use 2147483648 as offset and buffer
+          // store do nothing
+          Value oobAddrOp =
+              b.create<ConstantIntOp>(loc, 2147483647, b.getIntegerType(32));
```
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: define a constexpr or a const for the constant value

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h [hunk 11]
```diff
@@ -6771,17 +6918,124 @@ struct ThreadwiseCopyV2RewritePattern
       SmallVector<Value, 8> destLowerIndices =
           layeredDestIndices[layeredDestIndices.size() - 1];
 
+      // we do not  implement the logic of toEmitOOBStoreCheck yet
+      bool toEmitOOBStoreCheckLogic = false;
```
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h:7114]: Same as above. Please help follow changes recommended in threadwise_copy

**Review by whchung — CHANGES_REQUESTED** [review]

**Comment by whchung** [comment]
@asleepzzz commit e5579daf3a2942005423397d680b9a398601e030 shall be excluded from this PR as it had nothing to do with OOB store logic. It’s for generic padding kernels

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h [hunk 12]
```diff
@@ -2801,7 +2801,7 @@ struct Conv2DRewritePattern : public OpRewritePattern<T> {
 
     auto getGemmC = [&]() {
       // dim of oob ckeck
-      llvm::DenseSet<int> oobCheckDims;
+      llvm::DenseSet<int> oobLoadCheckDims;
```
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: is this truly `oobLoadCheckDims`? this is C matrix, so I assume it shall be `oobStoreCheckDims`?

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h [hunk 13]
```diff
@@ -6507,10 +6508,118 @@ struct ThreadwiseCopyRewritePattern
         SmallVector<Value, 8> destLowerIndices =
             layeredDestIndices[layeredDestIndices.size() - 1];
 
+        // we do not  implement the logic of toEmitOOBStoreCheck yet
+        bool toEmitOOBStoreCheckLogic = false;
+        SmallVector<unsigned, 2> oobStoreCheckDims;
+        // if (destLowerIndices.size() == 5) {
```
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: add "XXX" here to know it's experiment codes/comments here.

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h [hunk 14]
```diff
@@ -6507,10 +6508,118 @@ struct ThreadwiseCopyRewritePattern
...
+            Value upperBoundCheckOp = b.create<CmpIOp>(
+                loc, CmpIPredicate::slt, coordStore, upperBoundOp);
+            Value withinBoundInOneDimOp =
+                b.create<AndOp>(loc, lowerBoundCheckOp, upperBoundCheckOp);
+
+            withinStoreBoundsOp = b.create<AndOp>(loc, withinStoreBoundsOp,
+                                                  withinBoundInOneDimOp);
+            destLowerStoreOOBOp[dim] = zeroAddrOp;
+          }
+          // if you want to test it, use this code:
```
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: add "XXX" here to know it's experiment codes/comments here.

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h [hunk 15]
```diff
@@ -6771,17 +6880,125 @@ struct ThreadwiseCopyV2RewritePattern
       SmallVector<Value, 8> destLowerIndices =
           layeredDestIndices[layeredDestIndices.size() - 1];
 
+      // we do not  implement the logic of toEmitOOBStoreCheck yet
+      bool toEmitOOBStoreCheckLogic = false;
+      SmallVector<unsigned, 2> oobStoreCheckDims;
+      // if (destLowerIndices.size() == 5) {
```
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: add "XXX" here to know it's experiment codes/comments here.

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h [hunk 16]
```diff
@@ -6507,10 +6508,118 @@ struct ThreadwiseCopyRewritePattern
...
+
+          destLowerStoreOP.push_back(dstIndex0);
+          destLowerStoreOP.push_back(dstIndex1);
+          destLowerStoreOP.push_back(dstIndex2);
+          destLowerStoreOP.push_back(dstIndex3);
+          destLowerStoreOP.push_back(dstIndex4);
+          destLowerStoreOOBOp = destLowerStoreOP;
+          // 2G ,INT MAX Value = 2147483647, use 2147483648 as offset and buffer
+          // store do nothing
+          const int twoGB = 2147483647;
```
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: `twoGB` constant value is used between both `threadwise_copy` and `threadwise_copy_v2` so it's better to extract them to the beginning of the file.

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h (hunk 9 above)
Comment (whchung) [reply mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: these lines can be converted to a loop.

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h [hunk 17]
```diff
@@ -6771,17 +6880,125 @@ struct ThreadwiseCopyV2RewritePattern
...
-      } else if (dataType == b.getIntegerType(16)) {
-        auto convertValue =
-            b.create<miopen::DataConvertOp>(loc, dataType, scalarValue);
-        b.create<StoreOp>(loc, convertValue, op.dest(), destLowerIndices);
+      if (toEmitOOBStoreCheckLogic) {
+        auto zeroConstantOp = b.create<ConstantIndexOp>(loc, 0);
+        SmallVector<Value, 8> destLowerStoreOP;
+        SmallVector<Value, 8> destLowerStoreOOBOp;
+
+        auto dstIndex0 = b.create<IndexCastOp>(loc, destLowerIndices[0],
```
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: these lines can be converted to a loop.

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h [hunk 18]
```diff
@@ -6771,17 +6880,125 @@ struct ThreadwiseCopyV2RewritePattern
...
+          Value upperBoundCheckOp = b.create<CmpIOp>(loc, CmpIPredicate::slt,
+                                                     coordStore, upperBoundOp);
+          Value withinBoundInOneDimOp =
+              b.create<AndOp>(loc, lowerBoundCheckOp, upperBoundCheckOp);
+
+          withinStoreBoundsOp =
+              b.create<AndOp>(loc, withinStoreBoundsOp, withinBoundInOneDimOp);
+          destLowerStoreOOBOp[dim] = zeroAddrOp;
+        }
+        // if you want to test it, use this code:
```
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: add `XXX`

File: mlir/include/mlir/Dialect/GPU/GPUOps.td [hunk 19]
```diff
@@ -973,6 +973,7 @@ def GPU_MubufStoreOp :
                               VectorOfLengthAndType<[2, 4], [F32]>,
                               VectorOfLengthAndType<[2, 4, 8], [F16, I16]>]>:$value,
                    AnyMemRef:$memref,
+                   I32:$shift,
```
Comment (whchung) [inline mlir/include/mlir/Dialect/GPU/GPUOps.td:976]: please change the `description` of this op to briefly explain what this `shift` is. what should be the normal value, what should be the value when we want to do OOB store.

**Review by whchung — CHANGES_REQUESTED** [review]
additional changes to the codes and comments are needed.

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h (hunk 12 above)
Comment (asleepzzz) [reply mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: source index, suppoosed it's load?

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h [hunk 20]
```diff
@@ -2801,7 +2801,7 @@ struct Conv2DRewritePattern : public OpRewritePattern<T> {
 
     auto getGemmC = [&]() {
       // dim of oob ckeck
```
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: typo: ckeck -> check.

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h (hunk 12 above)
Comment (whchung) [reply mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: could you elaborate on how do you mean by "source index"?

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h [hunk 21]
```diff
@@ -6650,10 +6653,105 @@ struct ThreadwiseCopyRewritePattern
...
+        // }
+        //}
+        //}
+
         // Store to dest.
         // Issue scalar store.
-        b.create<StoreOp>(loc, convertedScalarValue, op.dest(),
-                          destLowerIndices);
+        if (toEmitOOBStoreCheckLogic) {
+          SmallVector<Value, 8> destLowerStoreOP;
```
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: nit: `destLowerStoreOP` -> `destLowerIndices` to have the same naming convention with load logic above.

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h [hunk 22]
```diff
@@ -6650,10 +6653,105 @@ struct ThreadwiseCopyRewritePattern
...
+          SmallVector<Value, 8> destLowerStoreOOBOp;
```
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: nit: `destLowerStoreOOBOp` -> `destLowerStoreOOBIndices` to have the same naming convention with load logic above.

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h [hunk 23]
```diff
@@ -2334,7 +2336,7 @@ struct Conv2DRewritePattern : public OpRewritePattern<T> {
 
     auto getGemmB = [&]() -> Value {
       // dim of oob check
-      llvm::DenseSet<int> oobCheckDims;
+      llvm::DenseSet<int> oobLoadCheckDims;
```
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: are we sure these dimensions are used for loads?

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h [hunk 24]
```diff
@@ -2802,8 +2804,8 @@ struct Conv2DRewritePattern : public OpRewritePattern<T> {
     };
 
     auto getGemmC = [&]() -> Value {
-      // dim of oob ckeck
-      llvm::DenseSet<int> oobCheckDims;
+      // dim of oob check
+      llvm::DenseSet<int> oobStoreCheckDims;
```
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: are we sure these dimensions are used for stores?

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h [hunk 25]
```diff
@@ -6650,10 +6653,105 @@ struct ThreadwiseCopyRewritePattern
         SmallVector<Value, 8> destLowerIndices =
             layeredDestIndices[layeredDestIndices.size() - 1];
 
+        // XXX: we do not  implement the logic of toEmitOOBStoreCheck yet
```
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: should we use `boundCheckDestAttr` , just like how `toEmitOOBLoadCheckLogic` is done with `boundCheckSourceAttr`?

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h (hunk 21 above)
Comment (asleepzzz) [reply mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h:6675]: SmallVector<Value, 8> destLowerIndices =
            layeredDestIndices[layeredDestIndices.size() - 1];

auto dstIndex0 = b.create<IndexCastOp>(loc, destLowerIndices[0],
                                                 b.getIntegerType(32));
destLowerStoreOP.push_back(dstIndex0);

Comment (asleepzzz) [reply mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h:6675]: destLowerCastIndices is better?

**Review by whchung — APPROVED** [review]
LGTM. Nice work!

Please inform @ltqin to review all lambdas added for backward data, and see if `bound_check` attributes are properly populated. In our earlier review it's identified that `gemmA` may have issues.

File: mlir/lib/Conversion/GPUToROCDL/LowerGpuOpsToROCDLOps.cpp [hunk 26]
```diff
@@ -362,7 +365,7 @@ struct MubufStoreOpLowering : ConvertToLLVMPattern {
     Value minusOne = rewriter.create<LLVM::ConstantOp>(
```
Comment (whchung) [inline mlir/lib/Conversion/GPUToROCDL/LowerGpuOpsToROCDLOps.cpp]: nit picking: remove unused code `minusOne` to remove compile time warnings.

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h [hunk 27]
```diff
@@ -6753,10 +6756,111 @@ struct ThreadwiseCopyRewritePattern
...
-        // Issue scalar store.
-        b.create<StoreOp>(loc, convertedScalarValue, op.dest(),
-                          destLowerIndices);
+        // Issue scalar store ,oob store only support f32 now
+        if (toEmitOOBStoreCheckLogic &&
+            destType.getElementType() == b.getF32Type()) {
+          SmallVector<Value, 8> destLowerStoreIndices;
+          SmallVector<Value, 8> destLowerStoreOOBIndices;
+
+          for (int i = 0; i < destLowerIndices.size(); i++) {
```
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: nit picking: use `unsigned` or `size_t` to remove compile time warnings. also it's a good practice in C++ to use `++i` instead of `i++`.

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h [hunk 28]
```diff
@@ -7017,17 +7127,118 @@ struct ThreadwiseCopyV2RewritePattern
...
-        auto convertValue =
-            b.create<miopen::DataConvertOp>(loc, dataType, scalarValue);
-        b.create<StoreOp>(loc, convertValue, op.dest(), destLowerIndices);
+      // oob store only support f32 now
+      if (toEmitOOBStoreCheckLogic &&
+          destType.getElementType() == b.getF32Type()) {
+        auto zeroConstantOp = b.create<ConstantIndexOp>(loc, 0);
+        SmallVector<Value, 8> destLowerStoreIndices;
+        SmallVector<Value, 8> destLowerStoreOOBIndices;
+        for (int i = 0; i < destLowerIndices.size(); i++) {
```
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: nit picking: use `unsigned` or `size_t` to remove compile time warnings. also it's a good practice in C++ to use `++i` instead of `i++`.

**Review by whchung — COMMENTED** [review]
Some additional cosmetic changes to help remove compile-time warnings.

**Comment by whchung** [comment]
@asleepzzz it seems CI node for `gfx908` is down right now. I'll take this PR once @okakarpa could bring it back online.

**Comment by whchung** [comment]
jenkins: retest this please.
